llvm/test/Transforms/IndVarSimplify/pr87798.ll

```diff
@@ -0,0 +1,36 @@
+; RUN: opt -S -passes='indvars' -verify-scev < %s | FileCheck %s
```

Comment:
Can you construct a testcase that uses -passes='print<scalar-evolution>'?  That would verify the underlying functionality is actually working correctly, instead of just not crashing.

Also, please don't mark tests `REQUIRES: asserts` unless it's actually hard required (e.g. the test uses `-debug`).

Reply:
Isn't `requires: assert` needed when the crash is an assertion failure since we want to show failure in DA/RA builds and it won't show up in release builds without assertions anyway?

Now, the requires won't be needed since I'll try adding the print SCEV pass.

Reply:
The test isn't supposed to crash whether or not assertions are enabled... running it with assertions disabled doesn't do any harm (and might catch related issues).

Reply:
Thanks, updated the test case.